packages/flutter_tools/test/general.shard/flutter_project_metadata_test.dart

```diff
@@ -184,4 +187,14 @@ migration:
 
     expect(logger.traceText, contains('The key `create_revision` was not found'));
   });
+
+  testUsingContext('enabledValues does not contain packageFfi if native-assets not enabled', () {
+    expect(FlutterProjectType.enabledValues, isNot(contains(FlutterProjectType.packageFfi)));
+  });
+
+  testUsingContext('enabledValues contains packageFfi if natives-assets enabled', () {
+    expect(FlutterProjectType.enabledValues, contains(FlutterProjectType.packageFfi));
+  }, overrides: <Type, Generator>{
+    FeatureFlags: () => TestFeatureFlags(isNativeAssetsEnabled: true),
+  });
```

Comment:
Can you also test a non-native assets feature flag to ensure it's added properly?